robocop/run.py

```diff
@@ -91,6 +113,7 @@ def run_checks(self):
                 checker.issues.clear()
             found_issues.sort()
             for issue in found_issues:
+                self.issues.append(issue)
```

Comment:
It would collect issues every time, even if we don't need it (for example for normal cli run). What about creating report that would collect issues, parsing them to json like you did? For that:
1) It would be better to keep reports as dict not a list so it's easier to access them (I can do that though, so we can stick to current implementation for now)
2) Create report inside reports 
```
class JsonReport(Report):
    def __init__(self):
        self.name = 'JsonReport'
        self.issues = []

    def add_message(self, message, **kwargs):  # pylint: disable=unused-argument
        self.issues.append(message.to_json()) # granted you add to_json() method to message
        
     def get_report(self):
         return ''
```

You can add then this method for retrieving issues to Robocop class:
```
def export_results(self): # not sure about naming
    for report in self.reports:
        if report.name == 'JsonReport':
            return report.issues
  
        
```

Reply:
There could be some gotchas or we would need to automatically register this report to get it working